opengrok-indexer/src/main/java/org/opengrok/indexer/history/SubversionHistoryParser.java

```diff
@@ -89,7 +102,12 @@ public void endElement(String uri, String localName, String qname) throws SAXExc
                 entry.setAuthor(s);
             } else if ("date".equals(qname)) {
                 try {
-                    entry.setDate(repository.parse(s));
+                    // need to strip microseconds off - assume final character is Z otherwise invalid anyway.
+                    String dateString = s;
+                    if (s.length() > 24) {
```

Comment:
Perhaps add e.g. `static final String SVN_MILLIS_DATE_EG` with the known-good format and then use `SVN_MILLIS_DATE_EG.length()` to avoid the unnamed literals here and the next line?

(This reminds of a TODO to refactor out of the old `DateFormat` for thread safety — and you've now made apparent also for microseconds handling.)

Reply:
Good idea.
Also thought it would be good to get rid of java.util.Date generally, but that seemed quite invasive for a first contribution!

Reply:
I assume you mean this comment: https://github.com/oracle/opengrok/blob/52fc2931f25089055b5cf1dea11f846521d748ef/opengrok-indexer/src/main/java/org/opengrok/indexer/history/Repository.java#L620

Having TODO's just in the source code is not good, filed  #3096 to track this.